harvester/harvester
# Remove deprecated beta.kubernetes.io/os label

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**


Harvester's Helm packaging references a deprecated label (beta.kubernetes.io/os). This is not causing any breakage, but it is resulting in a warning being logged.

**Solution:**


Remove the use of that label as a match selector. Harvester is already using its intended replacement (kubernetes.io/os), so just remove the deprecated one.

Before:

	$ helm upgrade --dry-run harvester . -n harvester-system | grep beta\.kubernetes\.io\/os | wc -l
	5

After:

	$ helm upgrade --dry-run harvester . -n harvester-system | grep beta\.kubernetes\.io\/os | wc -l
	1

Note that there is still 1 instance left and that will be the case until Harvester's helm chart pulls in a Helm release of harvester-network-controller that contains this commit: https://github.com/harvester/charts/commit/833cb089a53d38cdbd98b57659996430efe86668

**Related Issue:** https://github.com/harvester/harvester/issues/2899

**Test plan:**


One test is to _not_ apply this patch but to manually reproduce it:

```
$ kubectl edit deployment harvester -n harvester-system
```

1. Edit or delete the `management.cattle.io/scale-available` line.  (or any harmless line, this is just an example)
1. Save and exit your editor
1. Observe the warning:

> Warning: spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].key: beta.kubernetes.io/os is deprecated since v1.14; use "kubernetes.io/os" instead
deployment.apps/harvester edited

Now, remove the deprecated matchSelector:

```
$ kubectl edit deployment harvester -n harvester-system
```

```diff
-             - matchExpressions:
-                 - key: beta.kubernetes.io/os
-                   operator: In
-                   values:
-                     - linux
```

Save and exit your editor

Observe the error does not occur.
